### PR TITLE
[CT-1213] [Improvement] Update profiler to remove enabled arg (dbt-labs#5886)

### LINKED ISSUE #5886: [CT-1213] [Improvement] Update profiler to remove "enabled" arg
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

`dbt.profiler` is a context manager with an un-necessary "enabled" param.  The CM can simply be used conditionally rather than always adding the overhead of a CM.. I.E.

```
if <condition>:
    with profiler(outfile=<path_to_file>):
        stuff
```
instead of 

```
with profiler(enable=<conditional>, outfile=<path_to_file>):
```

This work should be done _sometime after the API-ification effort is merged into main_ (or at least it's own feature branch) as this pattern makes more sense in that context.

### Describe alternatives you've considered

_No response_

### Who will this benefit?

_No response_

### Are you interested in contributing this feature?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by johnmantios**
resolves #5886 

### Description

Removed redundant conditional statement in Profiler

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] Tests are not required/relevant for this PR
- [x] Docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: core/dbt/main.py
Comment (ChenyuLInx) [inline core/dbt/main.py:194]: This logic right now would leave current dbt not running `run_from_args` if profiler is not enabled

**Review by ChenyuLInx — CHANGES_REQUESTED** [review]
@johnmantios Thanks for contributing this!!
As the original issue pointed out, this change need to wait until work fore `API-ification` merged to main branch. That work is currently on [`feature/click-cli`](https://github.com/dbt-labs/dbt-core/tree/feature/click-cli).
After it is merged, the context manager we will be using will be supported by click `with_resource` method, then the `if profiler_enabled` logic can be expressed easier, for example, [this]( https://github.com/dbt-labs/dbt-core/blob/feature/click-cli/core/dbt/cli/main.py#L81) is how to have conditional context.

I think we can revisit this issue after feature branch is merged. Or try to open a PR against that feature branch.

File: core/dbt/main.py
Comment (johnmantios) [reply core/dbt/main.py:194]: Closed the PR to work on feature/click-cli branch instead of main

**Pull request closed without merging**
